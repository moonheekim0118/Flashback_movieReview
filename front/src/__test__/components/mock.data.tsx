export const MyInfoProp = 
{ 
    id:'test',
  nickname:'test',
  Reviews:1,
  profilePic:""
}  

export const MovieProp =  
{ 
    id:'test', 
    title:'test', 
    image:'test', 
    director:'test', 
    pubDate:'test'
};

export const AuthorProp = { id:'test', nickname:'test'};

export const ReviewProp = 
{ 
    id:'test', 
    Movie: MovieProp, 
    User:AuthorProp, 
    rating:'GOOD', 
    shortComment:'test',
    character:'test',
    line:'test',
    scene:'test',
    freeComment:'test',
};