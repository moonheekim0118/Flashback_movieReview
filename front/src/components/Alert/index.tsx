import React from 'react';
import { useSelector } from 'react-redux';
import styled from 'styled-components';


// 알림창 
const Alert=()=>{
    const { message, showAlert } = useSelector((state)=>state.alert);
    // showAlert 상태일때만 띄워준다. 

    return(
        <>
            {showAlert&&
            <Container>
                {message}
            </Container>}
        </>
    );
}

const Container = styled.div`
    position:fixed;
    left:50%;
    bottom:10%;
    transform:translateX(-50%);

    padding: 25px 15px;
    
    background-color:#cc00cc;
    border-raidus:10px;
    color:#fff;
    font-size:1.2rem;
    box-shadow:0px 0px 11px -1px rgba(0,0,0,0.75);

    z-index:7000;
`;

export default Alert;