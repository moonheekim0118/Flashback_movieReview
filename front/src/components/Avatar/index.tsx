import React from 'react';
import styled from 'styled-components';

interface Props {
    nickname?:string;
    imgSrc?:string;
    size?:number;
}

// 유저 아바타
const Avatar=({nickname, imgSrc="", size=40}:Props)=>{

    return(
        <Container size={size}>
            {imgSrc? <Image src={imgSrc}/> : <Nickname size={size}>{nickname[0]}</Nickname>}
        </Container>
    );
}

const Container = styled.div<{ size:number }>`
    width:${(props)=>`${props.size}px`};
    height:${(props)=>`${props.size}px`};
`;

const Image = styled.img`
    width:100%;
    height:100%;
    border-radius:100%;
    oveflow:hidden;
    object-fit:fill;
`;

const Nickname=styled.div<{ size:number }>`
    display:flex;
    justify-content:center;
    align-items:center;
    flex-direction:column;
    text-align:center;
    width:100%;
    height:100%;
    border-radius:100%;
    overflow:hidden;
    font-size:${(props)=>`${props.size-20}px`};
    color:#fff;
    background-color:#e0e0d1;
`;

export default Avatar;