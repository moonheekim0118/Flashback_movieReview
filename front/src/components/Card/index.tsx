import React from 'react';
import styled from 'styled-components';

interface Props {
  onClick?: (e: any) => void;
  posterSrc: string;
  alertChild?: React.ReactNode;
  descriptionChild: React.ReactNode;
  optionChild?: React.ReactNode;
}

const Card = ({
  onClick,
  posterSrc,
  alertChild,
  descriptionChild,
  optionChild,
}: Props) => {
  return (
    <Container onClick={onClick}>
      {alertChild}
      <MoviePoster src={posterSrc} alt="movie poster" />
      <MovieDescription>{descriptionChild}</MovieDescription>
      {optionChild}
    </Container>
  );
};

const Container = styled.article`
  display: flex;
  align-items: center;
  width: 100%;
  height: 200px;
  position: relative;

  border-bottom: 1px solid #f4f4f4;

  padding: 10px 20px;
  margin-top: 20px;
  cursor: pointer;
`;

const MoviePoster = styled.img`
  width: 150px;
  height: 100%;
  object-fit: scale-down;
  margin-right: 20px;
`;

const MovieDescription = styled.div`
  width: 100%;
  height: 50%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`;

export default Card;
