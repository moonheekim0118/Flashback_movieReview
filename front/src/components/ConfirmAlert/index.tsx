import React from 'react';
import ReactDOM from 'react-dom';
import styled from 'styled-components';

interface Props {
    text:string; // 알림창 메시지 
    clickYes:(e: any) => void; // 알림창에서 yes를 눌렀을 때 
    clickNo:(e: any) => void; // 알림차에서 No를 눌렀을 때 
}

// 확인 알림창
const ConfirmAlert=({text, clickYes, clickNo}:Props)=>{
    const root= document.getElementById('alert-root');
    
    return(
       ReactDOM.createPortal((
        <>
        <Overaly onClick={clickNo}/>
        <Container>
            <TextContainer>{text}</TextContainer>
            <ButtonContainer>
                <Button title="remove" id="submit" onClick={clickYes}>삭제</Button>
                <Button title="cancle" id="close" onClick={clickNo}>취소</Button>
            </ButtonContainer>
        </Container>
        </>
       ), root)
    );
}

const Overaly = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 8000;

  background-color: rgba(0, 0, 0, 0.3);
`


const Container = styled.div`
    display:flex;
    flex-direction:column;
    position:fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    width:310px;
    height:150px;
    
    background-color:${props => props.theme.colors.bgColor};
    color:${props => props.theme.colors.fontColor};
    border-raidus:10px;
    border: 1px solid #e0e0d1;
    font-size:1.2rem;

    z-index:9000;
`;

const TextContainer = styled.div`
    width:100%;
    height:65%;
    padding: 30px 0;
    text-align:center;
    border-bottom: 1px solid #e0e0d1;
`;
const ButtonContainer = styled.div`
    height:35%;
    width:100%;
`;

const Button = styled.button<{ title: string }>`
    border: none;
    border-right:${(props)=>props.title==='remove' ? '1px solid #e0e0d1;' : 'none'};
    width:50%;
    background-color:inherit;
    padding: 17px 10px;
    color:${(props)=>props.title==='remove' ? 'red' : 'green'};
    font-weight:bold;
    font-size:1rem;

    cursor:pointer;

    &:hover{
        background-color:#e0e0d1;
    }
`;

export default ConfirmAlert;