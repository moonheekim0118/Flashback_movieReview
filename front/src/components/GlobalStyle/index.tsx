import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
    *,
    *::after,
    *::before{
        box-sizing:border-box;
        margin:0;
        padding:0;
        font-family: 'Nanum Myeongjo', serif;
    }
    *:focus{
        outline:none;
    }
    body{
        overflow-x:hidden;
        color:#000000;
        background-color:#FFF;
    }
`;

export default GlobalStyle;
