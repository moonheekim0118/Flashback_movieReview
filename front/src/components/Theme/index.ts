export const lightTheme= {
   colors:{
       bgColor:'#fff',
       fontColor:'black'
   }
}


export const darkTheme= {
  colors:{
      bgColor:'#1a1a1a',
      fontColor:'#fff'
  }
}