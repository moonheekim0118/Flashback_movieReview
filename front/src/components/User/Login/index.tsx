import React , { useCallback } from 'react';
import styled from 'styled-components';
import Button from '../../../atoms/Buttons';
import useInput from '../../../hooks/useInput';


const Login=()=>{

    const [email, setEmail]=useInput("");
    const [password, setPassword]=useInput("");
    
    const onSubmit = useCallback((e)=>{
        e.preventDefault();
    },[email,password]);

    return(
       <Container>
           <InputContainer>
                <Label htmlFor="user-email">이메일</Label>
                <TextInput
                name="user-email"
                value={email}
                placeholder="example@example.com"
                onChange={setEmail}
                />     
           </InputContainer>  
           <InputContainer>
                <Label htmlFor="password">비밀번호</Label>
                <PasswordInput
                 name="user-password"
                 value={password}
                 placeholder="비밀번호 입력"
                 onChange={setPassword}
                />     
                
           </InputContainer>  
           <br/>
           <Button
            fill={true}
            shadow={true}
            disabled={email.length===0 || password.lenght===0}
            color={"purple"} 
            onClick={onSubmit}
            title={"로그인"}
           />
       </Container>
    );
}


export const Container = styled.form`
    width:75%;
    height:100%;
    margin:40px auto;
    padding-top:70px;
    
`;

export const InputContainer = styled.div`
    position:relative;
    display:flex;
    flex-direction:column;
    padding-bottom:50px;
`;

export const TextInput = styled.input.attrs({type:'text'})`
    width:100%;
    padding: 10px 15px;
    font-size:1.2rem;
`;

export const PasswordInput = styled.input.attrs({type:'password'})`
    width:100%;
    padding: 10px 15px;
    margin-top: 15px;

    font-size:1.2rem;
`;

export const Label = styled.label`
    font-size:1.2rem;
`;

export const ErrorMessage = styled.span`
    color:red;
    padding-top:10px;
    font-size:1rem;
`;

export default Login;