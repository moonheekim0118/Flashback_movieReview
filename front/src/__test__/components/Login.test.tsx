import 'jsdom-global/register'; 
import React from "react";
import { ThemeProvider } from 'styled-components';
import { lightTheme } from '../../components/Theme';
import { useSelector } from 'react-redux';
import { mount } from 'enzyme';
import Login from '../../components/User/Login';

jest.mock('react-redux', ()=>({
    useSelector : jest.fn(),
    useDispatch : jest.fn(()=>jest.fn()),
}));


describe('<Login/>',()=>{
    let container;
    beforeEach(()=>{
        const state = { loginError:false }; 
        useSelector.mockImplementation(()=>state);
        container = mount( 
        <ThemeProvider theme={lightTheme}>
            <Login/> 
        </ThemeProvider>
        )
    })

    it('should match correctly', ()=>{
        expect(container.html()).toMatchSnapshot();
    })
});