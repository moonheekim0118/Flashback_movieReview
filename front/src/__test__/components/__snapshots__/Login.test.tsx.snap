// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> should match correctly 1`] = `"<form class=\\"Login__Container-sc-1h2zokw-0 hoTOa-D\\"><div class=\\"Login__InputContainer-sc-1h2zokw-1 jPKskG\\"><label for=\\"user-email\\" class=\\"Login__Label-sc-1h2zokw-4 efwlLm\\">이메일</label><input name=\\"user-email\\" placeholder=\\"example@example.com\\" type=\\"text\\" class=\\"Login__TextInput-sc-1h2zokw-2 kGKXOg\\" value=\\"\\"></div><div class=\\"Login__InputContainer-sc-1h2zokw-1 jPKskG\\"><label for=\\"password\\" class=\\"Login__Label-sc-1h2zokw-4 efwlLm\\">비밀번호</label><input name=\\"user-password\\" placeholder=\\"비밀번호 입력\\" type=\\"password\\" class=\\"Login__PasswordInput-sc-1h2zokw-3 hDIsIn\\" value=\\"\\"></div><br><button fill=\\"true\\" color=\\"purple\\" disabled=\\"\\" class=\\"Buttons__Button-dz1ezt-0 hoQPBz\\">로그인</button><br></form>"`;
